captum/attr/_core/grad_cam.py

```diff
@@ -0,0 +1,157 @@
+                            tensors or any arbitrary python types. These arguments
+                            are provided to forward_func in order following the
+                            arguments in inputs.
+                            Note that attributions are not computed with respect
+                            to these arguments.
+                            Default: None
+
+            Returns:
+                *tensor* of **attributions**:
+                - **attributions** (*tensor*):
+                            Attributions based on GradCAM method.
+                            Attributions will be the same size as the
+                            output of the given layer, except for dimension 2,
+                            which will be 1 due to summing over channels.
+
+            Examples::
+
+                >>> # ImageClassifier takes a single input tensor of images Nx3x32x32,
+                >>> # and returns an Nx10 tensor of class probabilities.
+                >>> # It contains an attribute conv4, which is an instance of nn.conv2d,
+                >>> # and the output of this layer has dimensions Nx50x8x8.
+                >>> # It is the last convolution layer, which is the recommended
+                >>> # use case for GradCAM.
+                >>> net = ImageClassifier()
+                >>> layer_gc = LayerGradCam(net, net.conv4)
+                >>> input = torch.randn(2, 3, 32, 32, requires_grad=True)
+                >>> # Computes layer GradCAM for class 3.
+                >>> # attribution size matches layer output except for dimension
+                >>> # 1, so dimensions of attr would be Nx1x8x8.
+                >>> attr = layer_gc.attribute(input, 3)
+                >>> # GradCAM attributions are often upsampled and viewed as a
+                >>> # mask to the input, since the convolutional layer output
+                >>> # spatially matches the original input image.
+                >>> # This can be done with LayerAttribution's interpolate method.
+                >>> upsampled_attr = LayerAttribution.interpolate(attr, (32, 32))
+        """
+        inputs = format_input(inputs)
+        additional_forward_args = _format_additional_forward_args(
+            additional_forward_args
+        )
+        # Returns gradient of output with respect to
+        # hidden layer and hidden layer evaluated at each input.
+        layer_gradients, layer_eval = compute_layer_gradients_and_eval(
+            self.forward_func,
+            self.layer,
+            inputs,
+            target,
+            additional_forward_args,
+            device_ids=self.device_ids,
+        )
+        summed_grads = torch.mean(
+            layer_gradients,
+            dim=tuple(x for x in range(2, len(layer_gradients.shape))),
+            keepdim=True,
+        )
+
+        non_neg_scaled_act = F.relu(
+            torch.sum(summed_grads * layer_eval, dim=1, keepdim=True)
```

Comment:
This is a good point! I agree it would be nicer to give the user the flexibility to look at both positive and negative values. The only issue is that the paper did seem to imply that the relu is part of their description of GradCAM, so I'll add a note in the docstring that taking the positive of the returned attrs is necessary to match the original algorithm.